includes/wikia/models/NavigationModel.class.php

```diff
@@ -300,62 +263,55 @@ private function recursiveConvertMenuNodeToArray($index) {
 	 * @param string $source name of message / variable to be parsed
 	 * @param array $maxChildrenAtLevel allowed number of items on each menu level
 	 * @param int $duration cache duration
-	 * @param boolean $forContent use content language when parsing messages?
 	 * @param boolean $filterInactiveSpecialPages ignore item linking to not existing special pages?
+	 *
 	 * @return array parsed menu wikitext
 	 */
-	public function parse( $type, $source, Array $maxChildrenAtLevel = array(), $duration = 3600, $forContent = false, $filterInactiveSpecialPages = false ) {
-		wfProfileIn( __METHOD__ . ":$type");
-
-		$this->forContent = $forContent;
-
-		$cacheKey = $this->getMemcKey( $source );
-		if ( $forContent === false) {
-			$cacheKey .= ':forUserLang';
-		}
-
-		$nodes = $this->wg->Memc->get( $cacheKey );
-
-		if ( empty( $nodes ) ) {
-			wfProfileIn( __METHOD__  . '::miss' );
-
-			// get wikitext from given source
-			switch( $type ) {
-				case self::TYPE_MESSAGE:
-					$text = $this->forContent ? wfMsgForContent( $source ) : wfMsg( $source );
-					break;
-
-				case self::TYPE_VARIABLE:
-					// try to use "local" value
-					$text = $this->app->getGlobal( $source );
-
-					// fallback to WikiFactory value from community (city id 177)
-					if ( !is_string( $text ) ) {
-						$text = WikiFactory::getVarValueByName( $source, self::COMMUNITY_WIKI_ID );
-					}
-					break;
-				default:
-					$text = '';
-			}
-
-			// and parse it
-			$nodes = $this->parseText( $text, $maxChildrenAtLevel, $forContent, $filterInactiveSpecialPages );
-			$this->wg->Memc->set( $cacheKey, $nodes, $duration );
+	public function parse(
+		$type,
+		$source,
+		Array $maxChildrenAtLevel = [],
+		$duration = 3600,
+		$filterInactiveSpecialPages = false
+	) {
+		$nodes = WikiaDataAccess::cacheWithOptions(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$nodes". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)